crates/ruff_linter/src/rules/flake8_comprehensions/rules/unnecessary_literal_within_deque_call.rs

```diff
@@ -0,0 +1,122 @@
+use crate::checkers::ast::Checker;
+use ruff_diagnostics::{Diagnostic, Edit, Fix, FixAvailability, Violation};
+use ruff_macros::{derive_message_formats, ViolationMetadata};
+use ruff_python_ast::{self as ast, Expr};
+use ruff_text_size::Ranged;
+
+/// ## What it does
+/// Checks for usages of `collections.deque` that have an empty literal as the first argument.
+///
+/// ## Why is this bad?
+/// It's unnecessary to use an empty literal as a deque's iterable, since this is already the default behavior.
+///
+/// ## Examples
+/// ```python
+/// from collections import deque
+/// queue = deque(set())
+/// queue = deque([], 10)
+/// ```
+///
+/// Use instead:
+/// ```python
+/// from collections import deque
+/// queue = deque()
+/// queue = deque(maxlen=10)
+/// ```
+///
+/// ## References
+/// - [Python documentation: `collections.deque`](https://docs.python.org/3/library/collections.html#collections.deque)
+#[derive(ViolationMetadata)]
+pub(crate) struct UnnecessaryLiteralWithinDequeCall {
+    has_maxlen: bool,
+}
+
+impl Violation for UnnecessaryLiteralWithinDequeCall {
+    const FIX_AVAILABILITY: FixAvailability = FixAvailability::Always;
+
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        "Unnecessary literal within a deque call; remove the literal".to_string()
+    }
+
+    fn fix_title(&self) -> Option<String> {
+        let title = if self.has_maxlen {
+            "Replace with `deque(maxlen=...)`"
+        } else {
+            "Replace with `deque()`"
+        };
+        Some(title.to_string())
+    }
+}
+
+/// C421
+pub(crate) fn unnecessary_literal_within_deque_call(checker: &mut Checker, deque: &ast::ExprCall) {
+    let ast::ExprCall {
+        func, arguments, ..
+    } = deque;
+
+    let Some(qualified) = checker.semantic().resolve_qualified_name(func) else {
+        return;
+    };
+    if !matches!(qualified.segments(), ["collections", "deque"]) {
+        return;
+    }
+
+    let Some(iterable) = arguments
```

Comment:
I think we should bail if there are more than 2 arguments to avoid accidentally removing arguments.